Doc/library/typing.rst

```diff
@@ -627,7 +625,7 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
    Optional type.
 
-   ``Optional[X]`` is equivalent to ``Union[X, None]``.
+   ``Optional[X]`` is equivalent to ``X | None``.
```

Comment:
Can we mention the old way somewhere please? IMO, ``Union[X, None]`` was never deprecated, so it's still perfectly valid and a good alternative for people requiring multi-version support. We can just mention it with lower precedence

```suggestion
   ``Optional[X]`` is equivalent to ``X | None`` (or ``Union[X, None]``).
```